src/test/kotlin/com/terraformation/backend/accelerator/ProjectVariableSearchTest.kt

```diff
@@ -0,0 +1,249 @@
+package com.terraformation.backend.accelerator
+
+import com.terraformation.backend.RunsAsUser
+import com.terraformation.backend.TestClock
+import com.terraformation.backend.assertJsonEquals
+import com.terraformation.backend.customer.model.InternalTagIds
+import com.terraformation.backend.db.DatabaseTest
+import com.terraformation.backend.db.default_schema.Role
+import com.terraformation.backend.db.docprod.VariableType
+import com.terraformation.backend.i18n.Locales
+import com.terraformation.backend.i18n.use
+import com.terraformation.backend.mockUser
+import com.terraformation.backend.search.AndNode
+import com.terraformation.backend.search.FieldNode
+import com.terraformation.backend.search.NoConditionNode
+import com.terraformation.backend.search.SearchFieldPrefix
+import com.terraformation.backend.search.SearchResults
+import com.terraformation.backend.search.SearchService
+import com.terraformation.backend.search.table.SearchTables
+import io.mockk.every
+import java.math.BigDecimal
+import java.time.LocalDate
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+
+class ProjectVariableSearchTest : DatabaseTest(), RunsAsUser {
+  override val user = mockUser()
+
+  private val clock = TestClock()
+  private val searchService: SearchService by lazy { SearchService(dslContext) }
+  private val searchTables = SearchTables(clock)
+
+  @BeforeEach
+  fun setUp() {
+    insertOrganization()
+    insertOrganizationUser(
+        userId = inserted.userId, organizationId = inserted.organizationId, role = Role.Admin)
+    insertOrganizationInternalTag(tagId = InternalTagIds.Accelerator)
+
+    every { user.canReadAllAcceleratorDetails() } returns true
+    every { user.organizationRoles } returns mapOf(inserted.organizationId to Role.Admin)
+  }
+
+  @Test
+  fun `returns variables for project, ignoring old variables`() {
+    val projectId = insertProject()
+
+    val textStableId = "123"
+    val numberStableId = "456"
+    val dateStableId = "789"
+    val linkStableId = "101"
+    val oldVariableId = insertVariable(stableId = textStableId)
+    val newVariableId = insertVariable(stableId = textStableId, replacesVariableId = oldVariableId)
+    val otherVariableId = insertVariable(stableId = numberStableId, type = VariableType.Number)
+    val dateVariableId = insertVariable(stableId = dateStableId, type = VariableType.Date)
+    val oldLinkVariableId = insertVariable(stableId = linkStableId, type = VariableType.Link)
+    val newLinkVariableId =
+        insertVariable(
+            stableId = linkStableId,
+            type = VariableType.Link,
+            replacesVariableId = oldLinkVariableId)
+
+    // Add a connection between the project and each variable
+    insertValue(variableId = newVariableId, projectId = projectId, textValue = "NewVarOldVal")
```

Comment:
Could leave `projectId` out of these calls since it's the most recently inserted project which is the default.